### ISSUE TITLE
[Linux] Crash when playing video using custom libmpv.so

### ISSUE THREAD

**Issue by Predidit** [media-kit/media-kit#1051]
I am attempting to enable a custom libmpv.so on the Linux platform as mentioned in #1047 to eliminate dependency on the system mpv during both build and runtime. I have made some progress, and now the build process downloads the libmpv.so library and related header files hosted on GitHub. Now, media_kit_test can be built and run correctly on a device without mpv. However, I experience a crash when playing a video and receive the error: 

`media_kit_test: ../options/m_config_core.c:571: m_config_cache_from_shadow: asserts "group_index >= 0" failed`

I tried replacing my custom-built libmpv.so with the one from apt, but I still get this error.

@alexmercerind sorry to bother you, do you have any suggestions?

**Comment by Predidit** [comment]
I think I have fixed the issue mentioned above. To keep the issue more concise, I have removed the CMake-related code. Now, the only thing stopping me is the static compilation of libmpv.so. In fact, I already have a working libmpv.so, but it currently depends on many external libraries. I wonder if there is any precompiled static-linked libmpv.so binary or build script available. 

Also, although it may be a bit forward, I would like to know if we might merge this change if I continue working on it.

**Comment by alexmercerind** [comment]
Thanks a lot. I really appreciate your contribution. Unfortunately, my knowledge is also limited.

**Comment by mrRedSun** [comment]
@Predidit did you figure it out?

**Comment by Predidit** [comment]
@mrRedSun 

Yes, I have completed this work and you can use my branch. However, since the libmpv binaries come from another one of my repositories and include some out-of-tree patches, it’s currently difficult to upstream these changes.
